lib/jsonschema.lua

```diff
@@ -758,16 +758,38 @@ generate_validator = function(ctx, schema)
   if schema.minLength or schema.maxLength or schema.pattern then
     ctx:stmt(sformat('if %s == "string" then', datatype))
     if schema.minLength then
-      ctx:stmt(sformat('  if #%s < %d then', ctx:param(1), schema.minLength))
-      ctx:stmt(sformat('    return false, %s("string too short, expected at least %d, got %%d", #%s)',
-                       ctx:libfunc('string.format'), schema.minLength, ctx:param(1)))
-      ctx:stmt(        '  end')
+        ctx:stmt('        local c, j = 0, 1')
+        ctx:stmt(sformat('for i = 1, #%s do', ctx:param(1)))
+        ctx:stmt(sformat('  if j > #%s then break end', ctx:param(1)))
+        ctx:stmt(sformat('  local cb= string.byte(%s, j) ', ctx:param(1)))
```

Comment:
we need to use a local cache to avoid the global function.

`ctx:stmt(sformat('  local cb= %s(%s, j) ', ctx:libfunc('string.byte'), ctx:param(1)))`